PortalForecasts.R

```diff
@@ -144,20 +146,26 @@ forecastall <- function(abundances,level,weather_data,weatherforecast) {
   #Note: PI is missing. It has an error in the Poison Env model which produces NA values. 
   species=c('BA','DM','DO','DS','NA','OL','OT','PB','PE','PF','PH','PL','PM','PP','RF','RM','RO','SF','SH','SO','total')
   
+  #Model AIC sometimes doesn't work if species counts are low. In that case give a very large AIC.
   for(s in species) {
     species_abundance = abundances %>% 
       extract2(s)
 
     if(sum(species_abundance) == 0){
       pred = zero_abund_forecast
+      model_aic = 1e6
     } else {
       model=tsglm(species_abundance,model=list(past_obs=1,past_mean=12),distr="nbinom")
       pred=predict(model,12,level=0.9) 
+      model_aic = ifelse(has_error(summary(model))==T,1e6,summary(model)$AIC)
```

Comment:
If I'm understanding this line correctly, the `==T` is not necessary: if `has_error` returns `TRUE` or `FALSE`, then adding `==T` won't change its value (since `TRUE==T` returns `TRUE` and `FALSE==T` returns `FALSE`).

Also, using `T` as an abbreviation for `TRUE` is considered dangerous (especially in projects with multiple authors) because someone could set `T` to some other value (like the number of time steps) and break the code that relies on it.